cirq/circuits/circuit_dag.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2018 The ops Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, Generic, Iterator, Type, TypeVar
+
+import networkx
+
+from cirq import ops, devices
+from cirq.circuits import circuit
+
+
+T = TypeVar('T')
+
+class Unique(Generic[T]):
+    """A wrapper for a value that doesn't compare equal to other instances.
+
+    For example: 5 == 5 but Unique(5) != Unique(5).
+
+    Unique is used by CircuitDag to wrap operations because nodes in a graph
+    are considered the same node if they compare equal to each other.  X(q0)
+    in one moment of a Circuit and X(q0) in another moment of the Circuit are
+    wrapped by Unique(X(q0)) so they are distinct nodes in the graph.
+    """
+    def __init__(self, val: T) -> None:
+        self.val = val
+
+    def __repr__(self):
+        return 'Unique({}, {!r})'.format(id(self), self.val)
+
+
+TSelf = TypeVar('TSelf', bound='CircuitDag')
+
+
+def _default_can_reorder(op1: ops.Operation, op2: ops.Operation) -> bool:
```

Comment:
Call this "_disjoint_qubits" or similar so that people reading the documentation of the method using this as a default value have a description of what it does instead of how it's used.